### ISSUE TITLE
Feature request: optimistic mode for unannotated class files

### ISSUE THREAD

**Issue by msridhar** [typetools/checker-framework#2262]
This is following up on [a 2-year-old thread from the Checker Framework mailing list](https://groups.google.com/d/msg/checker-framework-discuss/JcX-rSbg1eI/bYPhquxXBgAJ).  Consider the following example:

`T1.java`:
```java
public class T1 {
    public static void checkStr(String s) {
      // uncommenting this line doesn't seem to make a difference,
      // but including just in case
      // s.toString();
    }
}
```
`T2.java`:
```java
public class T2 {
    public static void main(String[] args) {
        T1.checkStr(null);
    }
}
```
If we compile T1.java without the Checker Framework, and then run the Nullness Checker on when compiling T2, we get an error that the call `T1.checkStr(null)` is passing `@Nullable` where `@NonNull` is expected.  The manual does not make fully clear what behavior is expected for unannotated class files, but the thread linked above indicates that both parameters and return values are treated as `@NonNull` for such code.

It would be useful to have an "optimistic mode" for unannotated class files, where returns are assumed to be `@NonNull` but parameters are assumed to be `@Nullable`.  In fact, maybe that should be the default?

If the current default is desired, it would be great if it could be documented somewhere in the manual.

Possibly related to #12

**Comment by wmdietl** [comment]
Thanks for the comment and pointing out the documentation is lacking.

There is some description here: https://checkerframework.org/manual/#defaults-classfile

At the moment it isn't possible to easily switch between conservative and optimistic defaults for unchecked bytecode.

As type system designer one can use
https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/qual/DefaultInUncheckedCodeFor.java and
https://github.com/typetools/checker-framework/blob/master/framework/src/main/java/org/checkerframework/framework/qual/DefaultQualifierInHierarchyInUncheckedCode.java
to manually mark optimistic defaults (that is top qualifiers for parameters and bottom qualifiers for return types)
and then pass `-AuseDefaultsForUncheckedCode=bytecode` to use these defaults.

We should see whether we need optimistic, conservative, and custom as the three options for unchecked bytecode.
